src/main/java/me/zeroeightsix/kami/module/modules/player/PacketLogger.kt

```diff
@@ -3,98 +3,342 @@ package me.zeroeightsix.kami.module.modules.player
+                        }
+
+                        add(PacketSide.SERVER, it.packet, changedBlock)
+                    }
+                    is SPacketTimeUpdate -> {
+                        add(PacketSide.SERVER, it.packet,
+                            "totalWorldTime: ${it.packet.totalWorldTime} " +
+                            "worldTime: ${it.packet.worldTime}")
+                    }
+                    is SPacketChat -> {
+                        if (!ignoreChat) {
+                            add(PacketSide.SERVER, it.packet,
+                                "unformattedText: ${it.packet.chatComponent.unformattedText} " +
+                                "type: ${it.packet.type} " +
+                                "isSystem: ${it.packet.isSystem}")
+                        }
+                    }
+                    is SPacketTeams -> {
+                        add(PacketSide.SERVER, it.packet,
+                        "action: ${it.packet.action} " +
+                            "displayName: ${it.packet.displayName} " +
+                            "color: ${it.packet.color}")
+                    }
+                    is SPacketChunkData -> {
+                        add(PacketSide.SERVER, it.packet,
+                            "chunkX: ${it.packet.chunkX} " +
+                            "chunkZ: ${it.packet.chunkZ} " +
+                            "extractedSize: ${it.packet.extractedSize}")
+                    }
+                    is SPacketEntityProperties -> {
+                        add(PacketSide.SERVER, it.packet,
+                            "entityID: ${it.packet.entityId}")
+                    }
+                    is SPacketUpdateTileEntity -> {
+                        add(PacketSide.SERVER, it.packet,
+                            "posX: ${it.packet.pos.x} " +
+                            "posY: ${it.packet.pos.y} " +
+                            "posZ: ${it.packet.pos.z}")
+                    }
+                    is SPacketSpawnObject -> {
+                        add(PacketSide.SERVER, it.packet,
+                            "entityID: ${it.packet.entityID} " +
+                            "data: ${it.packet.data}")
+                    }
+                    is SPacketKeepAlive -> {
+                        if (!ignoreKeepAlive) {
+                            add(PacketSide.SERVER, it.packet,
+                                "id: ${it.packet.id}")
+                        }
+                    }
+                    else -> {
+                        if (!ignoreUnknown) {
+                            add(PacketSide.SERVER, it.packet, "Not Registered in PacketLogger.kt")
+                        }
+                    }
                 }
-            } catch (e: Exception) {
-                KamiMod.LOG.error("$chatName Error saving!", e)
             }
+        }
+
+        safeListener<PacketEvent.Send>(Int.MIN_VALUE) {
+            if (ignoreCancelled && it.cancelled) return@safeListener
```

Comment:
Do cancelled packets actually bubble up here? They used to not. Or is it because of the priority that it will now reach this listener?

Reply:
It means this listener will run *last*, ensuring anything cancelled on the Send is not logged

Reply:
Oh yes I see. That makes perfect sense thanks.